src/aggregator/aggregator/entry.go

```diff
@@ -646,9 +646,6 @@ func (e *Entry) updateStagedMetadatasWithLock(
 
 	// Update the metadatas.
 	for _, p := range sm.Pipelines {
-		if e.opts.AddToReset() {
```

Comment:
essentially we were transforming the map key only when adding, not when looking from the map (which happens here https://github.com/m3db/m3/blob/264b9eb34ae55a19ad44cf846a4e30cb93578c33/src/aggregator/aggregator/entry.go#L534)

Reply:
and we were efficiently allocating a new slice per metric ingested here

https://github.com/m3db/m3/blob/264b9eb34ae55a19ad44cf846a4e30cb93578c33/src/aggregator/aggregator/entry.go#L644

Reply:
and this idempotent check prevented any logic bugs (i.e creating duplicate aggs)

https://github.com/m3db/m3/blob/264b9eb34ae55a19ad44cf846a4e30cb93578c33/src/aggregator/aggregator/entry.go#L583